dependencies.yaml

```diff
@@ -57,16 +57,16 @@ dependencies:
           - cython>=0.29,<0.30
           - dlpack>=0.5,<0.6.0a0
           - ninja
-          - &pyarrow_ver pyarrow=9.0.0
+          - pyarrow=10.0.1
           - rmm=23.02.*
           - scikit-build>=0.13.1
       - output_types: conda
         packages:
-          - arrow-cpp=9
+          - arrow-cpp=10
```

Comment:
Use libarrow as above (and elsewhere).

Reply:
Done 👍  Was planning to do it a separate PR. But pushed it to this PR now, @ajschmidt8 do we need to change this elsewhere outside cudf? `arrow-cpp` is being replaced by `libarrow`

Reply:
Probably just this repo, based on search results. https://github.com/search?q=org%3Arapidsai%20arrow-cpp&type=code